projects/plugins/jetpack/sal/class.json-api-site-base.php

```diff
@@ -1348,15 +1348,23 @@ public function get_design_type() {
 	}
 
 	/**
-	 * Returns the 'siteGoals' option if set (eg. share, promote, educate, sell, showcase), null otherwise.
+	 * Returns the set site goals option ( siteGoals or site_goals ) if set (eg. share, promote, educate, sell, showcase).
 	 *
-	 * @return string|null
+	 * @return string|array|null
 	 **/
 	public function get_site_goals() {
 		$options = get_option( 'options' );
-		return empty( $options['siteGoals'] ) ? null : $options['siteGoals'];
-	}
 
+		if ( ! empty( $options['siteGoals'] ) ) {
+			return $options['siteGoals'];
+		}
+
+		if ( ! empty( $options['site_goals'] ) && is_array( $options['site_goals'] ) ) {
+			return $options['site_goals'];
+		}
+
+		return null;
+	}
```

Comment:
This is a breaking change. If somebody is consuming this function before your change, they'd expect a string or null, never an array. Now they will get an array and something may throw.

I did search though and nothing across Automattic is using this function at all. So maybe we should nuke `siteGoals` and keep only `site_goals: array|null`?

Reply:
Yeah it was mostly my paranoia about nuking something that I'm not 100% sure is not being used, being this Jetpack.


If we say that it isn't a problem, then let's go with nuking things!

Reply:
I was also trying to find where it is used but didn't. Looks like leftover code from way back

Reply:
Ok, cool 👌